### PR TITLE
feat: add evidence types and refactor db

### PR DESCRIPTION
**Pull request by moshloop**

### CODE REVIEW

File: db/migrations/004_incidents.sql [hunk 1]
```diff
@@ -137,13 +150,27 @@ CREATE TABLE evidences (
...
   hypothesis_id UUID NOT NULL,
+  config_id UUID NULL,
+  config_change_id UUID null,
+  config_analysis_id UUID null,
+  component_id UUID null,
+  check_id UUID null,
+  definition_of_done boolean, -- This indicates this item as needing to be fixed before closing the incident
+  done boolean, -- The evidence is done / resolved
+  factor boolean,
+  mitigator boolean,
```
Comment (yashmehrotra) [inline db/migrations/004_incidents.sql:161]: Should we store these in a json format ? Is there a possibility that more similar fields would be added ?

File: db/init.go [hunk 2]
```diff
@@ -82,17 +101,42 @@ func Init(connection string) error {
...
-	if err != nil {
+	if err = Migrate(); err != nil {
 		return err
 	}
 
-	return Migrate()
+	system := api.Person{}
+	if err = Gorm.Find(&system, "name = ?", "System").Error; err != nil {
+		return err
+	} else {
```
Comment (yashmehrotra) [inline db/init.go:135]: else not really required here

File: db/init.go [hunk 3]
```diff
@@ -82,17 +101,42 @@ func Init(connection string) error {
...
+		api.SystemUserID = &system.ID
+		logger.Infof("System user ID: %s", system.ID.String())
+	}
+	return nil
```
Comment (yashmehrotra) [inline db/init.go:139]:
```suggestion
	system := api.Person{}
	if err = Gorm.Find(&system, "name = ?", "System").Error; err != nil {
		return err
	}
	api.SystemUserID = &system.ID
	logger.Infof("System user ID: %s", system.ID.String())
	return nil
```

File: db/migrations/004_incidents.sql (hunk 1 above)
Comment (moshloop) [reply db/migrations/004_incidents.sql:161]: These are core fields used for incident lifecycle automation, and later for machine learning which means that would be queried alot

**Pull request closed without merging**